server/src/main/java/org/opensearch/gateway/ReplicaShardBatchAllocator.java

```diff
@@ -117,6 +119,7 @@ public AllocateUnassignedDecision makeAllocationDecision(ShardRouting unassigned
      * @param allocation    the allocation state container object
      */
     public void allocateUnassignedBatch(List<ShardRouting> shardRoutings, RoutingAllocation allocation) {
+        logger.debug("Starting shard allocation execution for unassigned replica shards: {}", shardRoutings.size());
```

Comment:
Please change this to trace level and capture the logs before and after the fix ?

Reply:
Done.